File: android/app/src/main/java/com/motorro/roompopulate/cities/data/City.kt
package com.motorro.roompopulate.cities.data

import androidx.room.Embedded
import androidx.room.Entity
import androidx.room.Index
import androidx.room.PrimaryKey
import com.neovisionaries.i18n.CountryCode

/**
 * City entity
 */
@Entity(
    tableName = "cities",
    indices = [
        Index(name = "city_name", value = ["name"])
    ]
)
data class City(
    @PrimaryKey
    val id: Int,
    val name: String,
    val state: String?,
    val country: CountryCode,
    @Embedded
    val coord: Coord
)

/**
 * Coordinates
 */
data class Coord(val lat: Double, val lon: Double)

File: script/cities/data/City.ts
/**
 * A city from city.list.json
 */
export interface City {
    readonly id: number
    readonly name: string
    readonly state: string
    readonly country: string
    readonly coord: Coord
}

/**
 * City coordinate
 */
export interface Coord {
    readonly lat: number
    readonly lon: number
}